UOP1_Project/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine;
+
+[RequireComponent(typeof(AudioSource))]
+public class AudioManager : MonoBehaviour
+{
+	private static AudioManager _instance;
+	private AudioSource _audioSource;
+
+	public static AudioManager Instance { get { return _instance; } }
+
+	private void Awake()
+	{
+		PreventMultipleAudioManagers();
+		_audioSource = GetComponent<AudioSource>();
+	}
+
+	public void Play(AudioClip audioClip)
+	{
+		_audioSource.clip = audioClip;
+		_audioSource.Play();
+	}
+
+	public void PlayOneShot(AudioClip audioClip)
+	{
+		_audioSource.PlayOneShot(audioClip);
+	}
+
+	public void Stop()
+	{
+		_audioSource.Stop();
+	}
+
+	private void PreventMultipleAudioManagers()
+	{
+		AudioManager[] otherAudioManagers = FindObjectsOfType<AudioManager>();
+		if (otherAudioManagers.Length > 1)
```

Comment:
Hey, another alternative is to check if the '_instance' is null. If the instance is null we know this is the first AudioManager, if it is not null we know there must alareay be an AudioManager in the scene. FindObjectsOfType can be quite an expensive operation where as null check is much quicker. 
```
if(_instance != null)
{
    gameObject.SetActive(false);
    Destroy(gameObject);
}
```